scipy/interpolate/tests/test_fitpack2.py

```diff
@@ -177,6 +178,21 @@ def test_increasing_x(self):
         # also test LSQUnivariateSpline [which needs explicit knots]
         spl = UnivariateSpline(xx, yy, check_finite=True)
         t = spl.get_knots()[3:4]  # interior knots w/ default k=3
+        spl2 = UnivariateSpline(x=x, y=y, w=w, check_finite=True)
+        spl3 = InterpolatedUnivariateSpline(x=x, y=y, check_finite=True)
+        spl4 = LSQUnivariateSpline(x=x, y=y, t=t, w=w, check_finite=True)
+    
+    def test_increasing_x(self):
+        # Test that x is required to be increasing, see ticket #8535
+        xx = np.arange(10, dtype=float)
+        yy = xx**3
+        x = np.arange(10, dtype=float)
+        x[1] = x[0] - 1.0
+        y = x**3
+        w = np.ones_like(x)
+        # also test LSQUnivariateSpline [which needs explicit knots]
+        spl = UnivariateSpline(xx, yy, check_finite=True)
```

Comment:
You don't seem to be using x, y, w for anything special. Maybe you can reuse `xx`, `yy`, `w` here ?